test/SchemaManager.Core.UnitTests/SqlSchemaManagerTests.cs

```diff
@@ -0,0 +1,105 @@
+        {
+            _client.GetCurrentVersionInformationAsync(Arg.Any<CancellationToken>()).ReturnsForAnyArgs(new List<CurrentVersion> { });
+
+            var current = await _sqlSchemaManager.GetCurrentSchema("connectionString", new Uri("https://localhost/"));
+
+            Assert.NotNull(current);
+            Assert.Empty(current);
+            await _baseSchemaRunner.ReceivedWithAnyArgs().EnsureBaseSchemaExistsAsync(default);
+            await _baseSchemaRunner.ReceivedWithAnyArgs().EnsureInstanceSchemaRecordExistsAsync(default);
+        }
+
+        [Fact]
+        public async Task GetAvailableSchema_SingleList_Succeeds()
+        {
+            _client.GetAvailabilityAsync(Arg.Any<CancellationToken>()).ReturnsForAnyArgs(new List<AvailableVersion> { new AvailableVersion(1, "_script/1.sql", "_script/1.diff.sql") });
+            var available = await _sqlSchemaManager.GetAvailableSchema(new Uri("https://localhost/"));
+
+            Assert.NotNull(available);
+            Assert.Single(available);
+            Assert.Equal(1, available[0].Id);
+            Assert.Equal("_script/1.sql", available[0].ScriptUri);
+            Assert.Equal("_script/1.diff.sql", available[0].DiffUri);
+        }
+
+        [Fact]
+        public async Task GetAvailableSchema_ContainsVersionZero_RemovesZero()
+        {
+            _client.GetAvailabilityAsync(Arg.Any<CancellationToken>()).ReturnsForAnyArgs(new List<AvailableVersion> { new AvailableVersion(0, "_script/0.sql", "_script/0.diff.sql"), new AvailableVersion(1, "_script/1.sql", "_script/1.diff.sql") });
+            var available = await _sqlSchemaManager.GetAvailableSchema(new Uri("https://localhost/"));
+
+            Assert.NotNull(available);
+            Assert.Single(available);
+            Assert.Equal(1, available[0].Id);
+            Assert.Equal("_script/1.sql", available[0].ScriptUri);
+            Assert.Equal("_script/1.diff.sql", available[0].DiffUri);
+        }
+
+        [Fact]
+        public async Task ApplySchema_UsingDiffScript_Succeeds()
+        {
+            _schemaManagerDataStore.GetCurrentSchemaVersionAsync(default).ReturnsForAnyArgs(Task.FromResult(1));
+            _client.GetCurrentVersionInformationAsync(Arg.Any<CancellationToken>()).ReturnsForAnyArgs(new List<CurrentVersion> { });
+            _client.GetAvailabilityAsync(Arg.Any<CancellationToken>()).ReturnsForAnyArgs(new List<AvailableVersion> { new AvailableVersion(1, "_script/1.sql", "_script/1.diff.sql"), new AvailableVersion(2, "_script/2.sql", "_script/2.diff.sql") });
+            _client.GetCompatibilityAsync(Arg.Any<CancellationToken>()).ReturnsForAnyArgs(new CompatibleVersion(1, 2));
+            _client.GetDiffScriptAsync(Arg.Is<Uri>(new Uri("_script/2.diff.sql", UriKind.Relative)), Arg.Any<CancellationToken>()).Returns("script");
+            await _sqlSchemaManager.ApplySchema("connectionString", new Uri("https://localhost/"), new MutuallyExclusiveType { Latest = true, Version = 2, Next = true });
```

Comment:
nit: There is a command line validation that only one of the option[version, latest, next] should be set in the object MutuallyExclusiveType.
So I would rather create - new MutuallyExclusiveType { Latest = false, Version = 2, Next = false}). So that one should not get confused, that we can input all at the same time.